code/modules/sports/baseball.dm

```diff
@@ -0,0 +1,97 @@
+/*
+	Bat procs
+*/
+/obj/item/weapon/material/twohanded/baseballbat/handle_shield(mob/user, var/damage, atom/damage_source = null, mob/attacker = null, var/def_zone = null, var/attack_text = "the attack")
+	if(user.incapacitated())
+		return 0
+
+	//block as long as they are not directly behind us
+	var/bad_arc = reverse_direction(user.dir) //arc of directions from which we cannot block
+	if(check_shield_arc(user, bad_arc, damage_source, attacker))
+		if(prob(get_block_chance(user, damage, damage_source, attacker)))
+			//if(istype(damage_source, /obj/item/weapon/ball/baseball)) ---------------------------------------------------------------------------------------
+			if(istype(damage_source, /obj/item/weapon))
+				user.visible_message("<span class='danger'>\The [user] hits [attack_text] with \the [src]!</span>")
+				return PROJECTILE_REFLECT
+	return 0
+
+/obj/item/weapon/material/twohanded/baseballbat/proc/get_block_chance(mob/user, var/damage, atom/damage_source = null, mob/attacker = null)
+	return base_block_chance
+
+/*
+	ball
+*/
+/atom/movable/proc/reflected(mob/user)
+	return
+
+/obj/item/weapon/ball/attack_self(mob/user as mob)
+	if(iscarbon(user))
+		var/mob/living/carbon/C = user
+		if(C.in_throw_mode)
+			C.throw_mode_off()
+		else
+			C.throw_mode_on()
+
+/*
+	Baseball
+*/
+/obj/item/weapon/ball/baseball
+	name = "baseball"
+	desc = "A ball."
+	icon = 'icons/obj/sports.dmi'
+	icon_state = "baseball"
+	w_class = 2
+
+/*
+	Reflection on being hit
+*/
+/obj/item/weapon/reflected(mob/user)
```

Comment:
It was intended to keep projectiles from being reflected so you couldn't use a baseball bat as a shield against a gun. The second part, no reason, just didn't think of it.